NugetLightRequest.cs

```diff
@@ -1718,24 +1862,32 @@ private IEnumerable<PackageItem> SearchForPackages(PackageSource source, string
                     return Enumerable.Empty<PackageItem>();
                 }
 
-                var isNameContainsWildCard = false;
                 var searchTerm = Contains.Value ?? string.Empty;
 
+                List<NuGetSearchTerm> searchTerms = new List<NuGetSearchTerm>();
                 // Deal with two cases here:
                 // 1. The package name contains wildcards like "Find-Package -name JQ*". We search based on the name.
                 // 2. A user does not provide the Name parameter,
 
                 // A user provides name with wildcards. We will use name as the SearchTerm while Contains and Tag will be used
                 // for filtering on the results.
-                if (!String.IsNullOrWhiteSpace(name) && WildcardPattern.ContainsWildcardCharacters(name)) 
+                StringBuilder searchTermsDebugMessageBuilder = new StringBuilder();
+                if (!String.IsNullOrWhiteSpace(name) && WildcardPattern.ContainsWildcardCharacters(name))
                 {
-                    isNameContainsWildCard = true;
+                    if (AllVersions.Value)
```

Comment:
any reasons we require allversions to be present?

Reply:
AllVersions.Value is false when not specified